composeApp/src/commonMain/kotlin/org/ooni/probe/domain/FetchGeoIpDbUpdates.kt

```diff
@@ -0,0 +1,171 @@
+package org.ooni.probe.domain
+
+import co.touchlab.kermit.Logger
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.withContext
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.SerializationException
+import kotlinx.serialization.json.Json
+import okio.FileSystem
+import okio.Path
+import okio.Path.Companion.toPath
+import org.ooni.engine.Engine
+import org.ooni.engine.Engine.MkException
+import org.ooni.engine.models.Failure
+import org.ooni.engine.models.Result
+import org.ooni.engine.models.Success
+import org.ooni.engine.models.TaskOrigin
+import org.ooni.probe.data.models.GetBytesException
+import org.ooni.probe.data.models.SettingsKey
+import org.ooni.probe.data.repositories.PreferenceRepository
+import kotlin.coroutines.CoroutineContext
+import kotlin.time.Clock
+
+class FetchGeoIpDbUpdates(
+    private val downloadFile: suspend (url: String, absoluteTargetPath: String) -> Result<Path, GetBytesException>,
+    private val cacheDir: String,
+    private val engineHttpDo: suspend (method: String, url: String, taskOrigin: TaskOrigin) -> Result<String?, Engine.MkException>,
+    private val preferencesRepository: PreferenceRepository,
+    private val json: Json,
+    private val fileSystem: FileSystem,
+    private val backgroundContext: CoroutineContext,
+) {
+    companion object {
+        private const val GEOIP_DB_VERSION_DEFAULT: String = "20250801"
+        private const val GEOIP_DB_REPO: String = "aanorbel/oomplt-mmdb"
+    }
+
+    suspend operator fun invoke(): Result<Path?, MkException> {
+        // Check if we've already checked today
+        val lastCheckMillis = preferencesRepository.getValueByKey(SettingsKey.MMDB_LAST_CHECK).first() as? Long
```

Comment:
Now that we have the check, we can prevent this check from happening in instrumented tests, like I have here with `disableRefreshArticles()`: https://github.com/ooni/probe-multiplatform/blob/6b3ab7199ab530d894c1bb374775b4a8c4cf7d9d/composeApp/src/androidInstrumentedTest/kotlin/org/ooni/probe/uitesting/helpers/StateTestHelpers.kt#L16